student_explorer/cron.py

```diff
@@ -149,8 +146,17 @@ def iterate_all_student_for_mentor(self, student_list, mentor_canvas_user_id ):
                 class_site_id = element.class_site.code
                 logger.info(f"class site id {class_site_id} {element.class_site}")
 
-                # if needed, add mentor as Observer to Canvas course site
-                self.add_user_as_observer_to_course(class_site_id, mentor_canvas_user_id, student_canvas_id)
+                # check whether mentor has been added into the course as student observer
+                qResult = MentorStudentCourseObserver.objects.filter(
+                            student_id=str(student_canvas_id),
+                            mentor_id = str(mentor_canvas_user_id),
+                            course_id = str(class_site_id))
+
+                if (len(qResult) == 0):
+                    # if needed, add mentor as Observer to Canvas course site
+                    self.add_user_as_observer_to_course(class_site_id, mentor_uniqname, mentor_canvas_user_id, student_canvas_id)
+                else:
+                    logger.info(f"STOP here: existing record in MentorStudentCourseObserver tracking table for student={student_canvas_id}, mentor={mentor_canvas_user_id}, course={class_site_id}")
```

Comment:
This probably should just be a debug message, it will print out often after all the users are successfully added.

Reply:
made it debug!